From aa817dce6bb0667fc6ef38163c213803dc8b37b3 Mon Sep 17 00:00:00 2001
From: "FeRD (Frank Dana)" <ferdnyc@gmail.com>
Date: Fri, 29 Sep 2023 04:23:22 -0400
Subject: [PATCH] Don't use deprecated pipes module

Only `pipes.quote()` was being used in Meld, and that symbol in
pipes is just an import of `shlex.quote()` anyway. Switch to calling
`shlex.quote()` directly, since pipes will be removed in Python 3.13.
---
 meld/melddoc.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/meld/melddoc.py b/meld/melddoc.py
index 8dd1a3376..27192f436 100644
--- a/meld/melddoc.py
+++ b/meld/melddoc.py
@@ -17,7 +17,6 @@
 import enum
 import logging
 import os
-import pipes
 import shlex
 import string
 import subprocess
@@ -45,7 +44,7 @@ def make_custom_editor_command(path: str, line: int = 0) -> Sequence[str]:
         log.error("Unsupported fields found")
         return [custom_command, path]
     else:
-        cmd = custom_command.format(file=pipes.quote(path), line=line)
+        cmd = custom_command.format(file=shlex.quote(path), line=line)
     return shlex.split(cmd)
 
 
-- 
GitLab

